taichi/analysis/data_source_analysis.cpp

```diff
@@ -37,6 +37,10 @@ std::vector<Stmt *> get_load_pointers(Stmt *load_stmt) {
     return external_func->arg_stmts;
   } else if (auto ref = load_stmt->cast<ReferenceStmt>()) {
     return {ref->var};
+  } else if (auto matrix_init = load_stmt->cast<MatrixInitStmt>()) {
+    return matrix_init->values;
+  } else if (auto ptr_offset = load_stmt->cast<PtrOffsetStmt>()) {
```

Comment:
Is this change to PtrOffsetStmt relevant to MatrixType?

Reply:
Yes, PtrOffsetStmt represents matrix indexing.

Reply:
Ah I see, we might migrate these PtrOffsetStmt to something like a MatrixPtrStmt, so it has more explicit semantic. But that's not urgent.